### PR TITLE
Write DG for Update Workout

### PR DESCRIPTION
**Pull request by TianaiYan**

### CODE REVIEW

**Review by a1021492980 — APPROVED** [review]
Good

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -491,30 +491,96 @@ rewrite all workouts to the resource file whenever a workout is deleted.
...
+
+**(Step 1 to 3)** When the `WorkoutCommand#execute()` method is called, 'workout /update' command is identified, and
+`WorkoutList#getCurrentWorkout()` will be called to get the name of the workout which will be updated later.
+
+Subsequently, `WorkoutList#updateWorkout()` method will be called.
+<br><br>
+
+The following sequence diagram is the detailed procedure for Step 4's `WorkoutList#updateWorkout()`:
+<br><br>
+![Delete Workout Detailed Sequence Diagram](uml/sequenceDiagrams/workouts/images/updateWorkout-Part2.png)
```
Comment (Musfirahe0556596) [inline docs/DeveloperGuide.md:524]: Might want to change the name of the text in between [ ] !

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -491,30 +491,96 @@ rewrite all workouts to the resource file whenever a workout is deleted.
...
+
+<span class="box info"> 🧾 To improve the diagram's readability, logging-related and input-checking method calls, and exception throws in
+ `WorkoutList#deleteWorkout()` have been omitted.</span>
```
Comment (Musfirahe0556596) [inline docs/DeveloperGuide.md:527]: Line 527: Is this supposed to be `updateWorkout()?` instead xD

**Review by Musfirahe0556596 — APPROVED** [review]
Looks good! Thanks for changing SD for deleteWorkout too!

**Review by alanlowzies — COMMENTED** [review]
Looks alright, just some suggestions for your consideration. Thank you :)

**Comment by Musfirahe0556596** [comment]
Fixes #205

**Pull request merged**